packages/grafana-ui/src/components/QueryField/QueryField.test.tsx

```diff
@@ -49,4 +49,36 @@ describe('<QueryField />', () => {
     expect(onBlur.mock.calls.length).toBe(1);
     expect(onRun.mock.calls.length).toBe(0);
   });
+  describe('syntaxLoaded', () => {
+    it('should re-render the editor after syntax has fully loaded', () => {
+      const wrapper: any = shallow(<QueryField query="my query" portalOrigin="mock-origin" />);
+      const spyOnChange = jest.spyOn(wrapper.instance(), 'onChange').mockImplementation(jest.fn());
+      wrapper.instance().editor = { insertText: () => ({ deleteBackward: () => ({ value: 'fooo' }) }) };
+      wrapper.setProps({ syntaxLoaded: true });
+      expect(spyOnChange).toHaveBeenCalledWith('fooo', true);
+    });
+    it('should not re-render the editor if syntax is already loaded', () => {
+      const wrapper: any = shallow(<QueryField query="my query" portalOrigin="mock-origin" />);
+      const spyOnChange = jest.spyOn(wrapper.instance(), 'onChange').mockImplementation(jest.fn());
+      wrapper.setProps({ syntaxLoaded: true });
+      wrapper.instance().editor = {};
+      wrapper.setProps({ syntaxLoaded: true });
+      expect(spyOnChange).not.toBeCalled();
+    });
+    it('should not re-render the editor if editor itself is not defined', () => {
+      const wrapper: any = shallow(<QueryField query="my query" portalOrigin="mock-origin" />);
+      const spyOnChange = jest.spyOn(wrapper.instance(), 'onChange').mockImplementation(jest.fn());
+      wrapper.setProps({ syntaxLoaded: true });
+      expect(wrapper.instance().editor).toBeFalsy();
+      expect(spyOnChange).not.toBeCalled();
+    });
+    it('should not re-render the editor twice once syntax is fully lodaded', () => {
+      const wrapper: any = shallow(<QueryField query="my query" portalOrigin="mock-origin" />);
+      const spyOnChange = jest.spyOn(wrapper.instance(), 'onChange').mockImplementation(jest.fn());
+      wrapper.instance().editor = { insertText: () => ({ deleteBackward: () => ({ value: 'fooo' }) }) };
+      wrapper.setProps({ syntaxLoaded: true });
+      wrapper.setProps({ syntaxLoaded: true });
```

Comment:
Unnecessary line repeat?

Reply:
No, actually is testing whether the logic inside ```if``` is executed only once, although I'm not sure how meaningful it is. Maybe in the future, if someone changes the logic by mistake this test will complain.